src/ILCompiler.ReadyToRun/src/ILCompiler.ReadyToRun.csproj

```diff
@@ -14,6 +15,7 @@
     <ProjectReference Include="..\..\ILCompiler.MetadataWriter\src\ILCompiler.MetadataWriter.csproj" />
     <ProjectReference Include="..\..\ILCompiler.TypeSystem\src\ILCompiler.TypeSystem.csproj" />
     <ProjectReference Include="..\..\ILCompiler.Compiler\src\ILCompiler.Compiler.csproj" />
+    <ProjectReference Include="..\..\ILCompiler.RyuJit\src\ILCompiler.RyuJit.csproj" />
```

Comment:
Why does the CoreCLR R2R flavor depends on the CoreRT RyuJIT flavor? Can we get away without this reference? It seems like something that will get into way, eventually.

Reply:
We derive from `MethodCodeNode` which I moved to RyuJit and also need IMethodCodeNode. I could link them in to the ReadyToRun project and kill the reference to `ILCompiler.RyuJit` if you think that's a better approach.

Reply:
> We derive from MethodCodeNode which I moved to RyuJit

You mean `MethodWithGCInfo` in Tomas's prototype? Seems like that already has different logic in it to represent GC info and it might diverge even more. Do we really need to derive from `MethodCodeNode`?

`IMethodCodeNode` is something that was added for the JIT prototype - I'm hoping that with making `CorInfoImpl` a partial class, we'll be able to get rid of this interface.

Maybe the reference to ILCompiler.RyuJIT is fine for now, but let's keep a list of things to undo somewhere: killing `IMethodCodeNode`, stopping `MethodWithGCInfo` from deriving from `MethodCodeNode`, and removing this reference should be on the list.

Reply:
It looks like we can get rid of `IMethodCodeNode` pretty easily since its use in CorInfoImpl is very minimal. I haven't experimented with removing the inheritance between `MethodCodeNode` and `MethodWithGCInfo`. I think this set of changes is enough to commit as-is. We can track further cleanup (including JitHelper.cs) in an issue for follow up work.

Reply:
See https://github.com/dotnet/corert/issues/6115